src/components/MultiHopTrade/components/LimitOrder/components/LimitOrderInput.tsx

```diff
@@ -238,26 +216,88 @@ export const LimitOrderInput = ({
   // input, and then re-attached to the `LimitOrder` before signing and submitting via our
   // `placeLimitOrder` endpoint in limitOrderApi
   const {
-    data,
+    data: quoteResponse,
     error,
     isFetching: isLimitOrderQuoteFetching,
   } = useQuoteLimitOrderQuery(limitOrderQuoteParams)
 
   const marketPriceBuyAsset = useMemo(() => {
     // RTK query returns stale data when `skipToken` is used, so we need to handle that case here.
-    if (!data || limitOrderQuoteParams === skipToken) return '0'
+    if (!quoteResponse || limitOrderQuoteParams === skipToken) return '0'
 
-    return bnOrZero(fromBaseUnit(data.quote.buyAmount, buyAsset.precision))
-      .div(fromBaseUnit(data.quote.sellAmount, sellAsset.precision))
+    return bnOrZero(fromBaseUnit(quoteResponse.quote.buyAmount, buyAsset.precision))
+      .div(fromBaseUnit(quoteResponse.quote.sellAmount, sellAsset.precision))
       .toFixed()
-  }, [buyAsset.precision, data, sellAsset.precision, limitOrderQuoteParams])
+  }, [buyAsset.precision, quoteResponse, sellAsset.precision, limitOrderQuoteParams])
 
   // Reset the limit price when the market price changes.
   // TODO: If we introduce polling of quotes, we will need to add logic inside `LimitOrderConfig` to
   // not reset the user's config unless the asset pair changes.
   useEffect(() => {
-    setLimitPriceBuyAsset(marketPriceBuyAsset)
-  }, [setLimitPriceBuyAsset, marketPriceBuyAsset])
+    setLimitPrice({
+      [PriceDirection.BuyAssetDenomination]: marketPriceBuyAsset,
+      [PriceDirection.SellAssetDenomination]: bn(1).div(marketPriceBuyAsset).toFixed(),
+    } as Record<PriceDirection, string>)
+  }, [marketPriceBuyAsset, setLimitPrice])
+
+  const onSubmit = useCallback(() => {
+    // No preview happening if wallet isn't connected i.e is using the demo wallet
+    if (!isConnected || isDemoWallet) {
+      return handleConnect()
+    }
+
+    // RTK query returns stale data when `skipToken` is used, so we need to handle that case here.
+    // This should never happen because we're meant to disable the confirmation button when this is
+    // true, but just in case.
+    if (!quoteResponse || limitOrderQuoteParams === skipToken || !sellAccountId) {
+      setActiveQuote(undefined)
+      return
+    }
+
+    setIsConfirmationLoading(true)
```

Comment:
q: don't you want to put this useCallback logic into a `useMutation` so the loading logic is included

Reply:
Well spotted and actually we can remove these setters entirely, this is an artefact of the initial skeleton implementation where we were expecting async logic which didnt end up existing

Reply:
2b29f316c6f238ce94a68748d1a03fe01cdd5c2a